app/grandchallenge/direct_messages/templates/direct_messages/conversation_detail.html

```diff
@@ -0,0 +1,156 @@
+                                      hx-target="#conversation-detail-panel">
+                                    {% csrf_token %}
+                                    <input type="hidden" name="source" value="{{ request.user.pk }}" disabled="">
+                                    <input type="hidden" name="target" value="{{ participant.pk }}" disabled="">
+                                    <input type="hidden" name="conversation" value="{{ object.pk }}">
+                                    <button class="dropdown-item"
+                                            type="submit"
+                                            value="Block"><i class="fas fa-comment-slash text-danger"></i> Block User
+                                    </button>
+                                </form>
+                            {% endif %}
+                        </div>
+                    </div>
+                </div>
+            </div>
+        {% endif %}
+    {% endfor %}
+</div>
+
+<div id="conversation-direct-messages-panel"
+     class="card-body overflow-auto flex-grow-1"
+     data-conversation-select-detail-url="{% url 'direct-messages:conversation-select-detail' pk=object.pk %}"
+     data-conversation-select-button-selector="#conversation-select-button-{{ object.pk }}">
+    {% for direct_message in object.direct_messages.all %}
+        {% ifchanged direct_message.created.date %}
+            <hr>
+            <div class="d-flex w-100 justify-content-center">
+                <h6>{{ direct_message.created.date }}</h6>
+            </div>
+        {% endifchanged %}
+
+        {% ifchanged direct_message.sender direct_message.created.date %}
+            <div class="d-flex w-100 justify-content-between">
+                <h6>
+                    <img class="rounded-circle align-middle"
+                         src="{{ direct_message.sender.user_profile.get_mugshot_url }}"
+                         alt="User Mugshot"
+                         style="height: 1.33em"/>&nbsp;
+                    {% if direct_message.sender in muted_users %}<s>{% endif %}
+                    <b>{{ direct_message.sender.first_name }}&nbsp;{{ direct_message.sender.last_name }}</b>&nbsp;(<a href="{{ direct_message.sender.user_profile.get_absolute_url }}">{{ direct_message.sender.username }}</a>)
+                    {% if direct_message.sender in muted_users %}</s>{% endif %}
+                </h6>
+                <span>{{ direct_message.created|time }}</span>
+            </div>
+        {% endifchanged %}
+
+        <div class="d-flex w-100 justify-content-start">
+            {% if direct_message.unread_by_user %}
+                <span class="text-danger"><i class="fa fa-xs fa-circle"></i></span>&nbsp;
+            {% endif %}
+
+            <div>
+                {% include "direct_messages/partials/directmessage_detail.html" with preview=False %}
+            </div>
```

Comment:
As I was trying things out locally, I found it a bit difficult to tell my own messages apart from those of other people in the conversation. We could right align the users messages and left align the rest (see whatsapp and other message apps). Idk, just a thought.

Reply:
Indeed, we can change it in the future, I think this is OK for the initial version.